samples/boards/reel_board/mesh_badge/src/mesh.c

```diff
@@ -352,9 +354,46 @@ static void vnd_heartbeat(struct bt_mesh_model *model,
 	board_add_heartbeat(ctx->addr, hops);
 }
 
+static size_t first_name_len(const char *name)
+{
+	size_t len;
+
+	for (len = 0; *name; name++, len++) {
+		switch (*name) {
+		case ' ':
+		case ',':
+		case '\n':
+			return len;
+		}
+	}
+
+	return len;
+}
+
+static void vnd_get_name(struct bt_mesh_model *model,
+			  struct bt_mesh_msg_ctx *ctx,
+			  struct net_buf_simple *buf)
+{
+	const char *name = bt_get_name();
+	NET_BUF_SIMPLE_DEFINE(msg, 3 + HELLO_MAX + 4);
+
+	/* Ignore messages from self */
+	if (ctx->addr == bt_mesh_model_elem(model)->addr) {
+		return;
+	}
+
+	bt_mesh_model_msg_init(&msg, OP_VND_GET_NAME);
```

Comment:
This looks very strange to me. You're sending "Get Name" as a response to a "Get Name" request?